### ISSUE TITLE
Improper handling of endpoint when connecting to voice

### ISSUE THREAD

**Issue by Techtony96** [austinv11/Discord4J#298]
### Prerequisites
* [X] If this is a question/suggestion, I have already considered talking about it on the [Discord4J server](https://discord.gg/NxGAeCY)
* [X] This issue specifically has something to do with Discord4J
* [X] I have attempted to [look for similar issues](https://github.com/austinv11/Discord4J/issues?utf8=%E2%9C%93&q=is%3Aissue) already

### Description
D4J does not handle the awaiting endpoint status from discord

### Steps to Reproduce
1. Be on latest dev
2. try to connect to a voice channel

...

**Expected behavior:** You are successfully connected to the voice channel

**Actual behavior:** And NPE was thrown because the endpoint is null

**Stacktrace (if applicable):** NPE on DiscordVoiceWS.java:65

**Version affected:** dev-snapshot

### Additional Information
[Any other information that may be able to help me with the problem]

**Comment by MinnDevelopment** [comment]
A null endpoint can happen when discord needs to set up resources for a connection. They will send an additional VOICE_SERVER_UPDATE dispatch with an endpoint when the server is ready.

**Comment by Techtony96** [comment]
So, it turns out that my code is the one forcing my bot to disconnect as soon as it is connected to a voice channel, and is not a problem with the handling of the endpoint. 

But, there seems to be a change in the way the `UserVoiceChannelLeaveEvent` Event is being handled between 2.8.0 (not sure about 2.8.1) and the dev branch. I'm going to walk you though what my code does and we can figure out if this is the new expected behavior or a bug.

1. I tell my bot to join the channel I am currently in with !Join.
2. The bot connects and registers some events (`UserVoiceChannelLeaveEvent`, `GuildLeaveEvent`, and `UserVoiceChannelMoveEvent`) to automatically disconnect if the bot is all alone.
3. The `UserVoiceChannelLeaveEvent` event is fired when my bot joins. (this seems to be problem number one to me)
4. I call a method to check the connected users, and if it is <= 1, disconnect. This is true, even though the bot and myself are connected. (Only myself is shown as a connected user) (Problem number 2?)
5. The bot leaves.

So in summary, when the bot joins a voice channel, two different behaviors happen:
2.8.0: `UserVoiceChannelLeaveEvent` is NOT fired when the bot joins
dev:  `UserVoiceChannelLeaveEvent` is fired on join.
